mmv1/products/networksecurity/SecurityProfile.yaml

```diff
@@ -0,0 +1,176 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'SecurityProfile'
+base_url: '{{parent}}/locations/{{location}}/securityProfiles'
+create_url: '{{parent}}/locations/{{location}}/securityProfiles?securityProfileId={{name}}'
+self_link: '{{parent}}/locations/{{location}}/securityProfiles/{{name}}'
+min_version: beta
+update_verb: :PATCH
+update_mask: true
+description: |
+  A security profile defines the behavior associated to a profile type.
+references:
+  !ruby/object:Api::Resource::ReferenceLinks
+  api: 'https://cloud.google.com/firewall/docs/reference/network-security/rest/v1beta1/projects.locations.securityProfiles'
+  guides:
+    'Create and manage security profiles': 'https://cloud.google.com/firewall/docs/configure-security-profiles'
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    base_url: '{{op_id}}'
+import_format: ['{{%parent}}/locations/{{location}}/securityProfiles/{{name}}']
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'network_security_security_profile_basic'
+    primary_resource_id: 'default'
+    min_version: beta
+    vars:
+      resource_name: 'my-security-profile'
+    test_env_vars:
+      org_id: :ORG_ID
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'network_security_security_profile_overrides'
+    primary_resource_id: 'default'
+    min_version: beta
+    vars:
+      resource_name: 'my-security-profile'
+    test_env_vars:
+      org_id: :ORG_ID
+custom_code:
+  !ruby/object:Provider::Terraform::CustomCode # Calling custom operation that are different from other network security resources.
+  pre_create: templates/terraform/pre_create/network_security_security_profile_operation.go.erb
+  pre_update: templates/terraform/pre_update/network_security_security_profile_operation.go.erb
+  pre_delete: templates/terraform/pre_delete/network_security_security_profile_operation.go.erb
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The name of the security profile resource.
+  - !ruby/object:Api::Type::String
+    name: 'location'
```

Comment:
Just checking, is location mutable?

Reply:
Being part of the resource id itself, I'd assume yes for now. Also, there's nothing obvious in the API definition that lets me think otherwise. I'll make it immutable.